### DISCUSSION TITLE
Cannot set an optional foreign key to null in Prisma

### DISCUSSION THREAD

**Discussion by tester2-austr** [prisma/prisma#19110] [Q&A] [answered]
### Bug description

I am trying to set an an optional foreign key  `Vehicle_Model_Category` in the `Vehicle` model to null because I have set as optional, but when I try to set as null it gives me an error  that 
`Argument Vehicle_Model_Category for data.Vehicle_Model_Category must not be null. Please use undefined instead.`

### How to reproduce

If you want to reproduce try to run the two schema and try to update a vehicle and setting to null

### Expected behavior

The expected behavior is:
`Argument Vehicle_Model_Category for data.Vehicle_Model_Category must not be null. Please use undefined instead.`

### Prisma information

```prisma
 model Vehicle {
          id                        String                  @id @default(uuid())
          vehicle_model_category_id String?
          Vehicle_Model_Category    Vehicle_Model_Category? @relation(fields: [vehicle_model_category_id], references: [id], onUpdate: Cascade, onDelete: Cascade)
     }
     
  model Vehicle_Model_Category {
  id                      String               @id @default(uuid())
  name                    String               @db.VarChar(255)
 ....
  Vehicle                 Vehicle[]
}
```

```ts
   const vehicle = await prisma.vehicle.update({
      where: {
        id,
      },

      data:{
      Vehicle_Model_Category: null,
      }
 });

```

### Environment & setup

- OS: Windows
- Database: MySQL
- Node.js version: v16.15.1

### Prisma Version

```
prisma                  : 4.2.1
@prisma/client          : 4.13.0
```

**Comment by nurul3101** [comment]
Hello @p403-hash 👋 
Can you try this query and check?
```ts
   const vehicle = await prisma.vehicle.update({
      where: {
        id,
      },
      data:{
        vehicle_model_category_id: null,
      }
 });
```
You can also use [disconnect](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#disconnect-a-related-record) to remove an existing connected record.

**Reply by ParidV** [reply, marked as answer]
Hey, I think this will work
```
const vehicle = await prisma.vehicle.update({
    where: {
        id,
    },
    data: {
        Vehicle_Model_Category: {
            {
                disconnect: true,
            }
        }

    }
});
```
